src/mdlComponentHandler.js

```diff
@@ -389,7 +374,15 @@ componentHandler = (function() {
      * @param  {!Node} node the node to be downgraded
      */
     var downgradeNode = function(node) {
-      deconstructComponentInternal(findCreatedComponentByNodeInternal(node));
+      var found = [];
+      createdComponents_.forEach(function(item) {
+        if (item.element_ === node) {
+          found.push(item);
```

Comment:
can we downgrade within the loop?

reatedComponents_.forEach(function(item) {
  ...
  deconstructComponentInternal(item);
  ...